lib/player.js

```diff
@@ -3796,70 +3798,116 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
    * @param {string=} mimeType
    * @param {string=} codec
    * @param {string=} label
-   * @return {shaka.extern.Track}
+   * @return {!Promise.<shaka.extern.Track>}
    * @export
    */
-  addTextTrack(uri, language, kind, mimeType, codec, label) {
-    // TODO: Add an actual error for this.
+  async addTextTrack(uri, language, kind, mimeType, codec, label) {
+    if (this.loadMode_ != shaka.Player.LoadMode.MEDIA_SOURCE &&
+        this.loadMode_ != shaka.Player.LoadMode.SRC_EQUALS) {
+      shaka.log.error(
+          'Must call load() and wait for it to resolve before adding text ' +
+          'tracks.');
+      return Promise.reject(() => {
+        return new shaka.util.Error(
+            shaka.util.Error.Severity.RECOVERABLE,
+            shaka.util.Error.Category.PLAYER,
+            shaka.util.Error.Code.PLAYER_UNINITIALIZED);
+      });
```

Comment:
There are two things wrong here:

1. The value you are giving to the rejection is a function that returns an Error, but the value should be the Error itself.
2. Stylistically, you should not use Promise.reject in an async function.  Instead, just throw the Error value, and the returned Promise from the async function will be rejected with that value.